Misc/NEWS.d/next/Library/2022-10-14-19-57-37.gh-issue-96035.0xcX-p.rst

```diff
@@ -0,0 +1,3 @@
+Fix bug in :func:`urllib.parse.urlparse` that causes certain port numbers
+containing whitespace, underscores, or plus and minus signs to be
```

Comment:
```suggestion
containing whitespace, underscores, plus and minus signs, or non-ASCII digits to be
```

Reply:
This may have some crazy implications if there are sequences of non-ASCII digits that contain 0x0d, 0x0a.
Or 0x2f, which would start a path segment.

Reply:
In UTF-8 multibyte characters always start with at least one high 0x80 bit set on every byte so this is not possible unless this code is expected to process `bytes` in arbitrary unknown encodings (which really isn't feasible and should blow up in other places if such bytes were given as input).